### DISCUSSION TITLE
CSV select delimiter

### DISCUSSION THREAD

**Discussion by MixusMinimax** [meilisearch/product#535] [Feedback & Feature Proposal]
Currently, the function `milli::documents::builder::from_csv` uses the default csv Reader. It would be very nice if the function accepted either an entire `ReaderBuilder`, or at least a delimiter.
`DocumentAdditionFormat` could then be:
```rust
#[derive(Debug)]
pub enum DocumentAdditionFormat {
    Json,
    Csv(u8),
    Ndjson,
}
```
so then the payload you pass to `MeiliSearch::register_update` can use a different delimiter.
I currently sadly have to parse my csv, then serialize it back down, or at least do some kind of string manipulation on the source csv.

Edit: In the latest version of milli (not used by the latest version of meilisearch), it already accepts a csv reader. So it only needs to be changed in [meilisearch-lib/src/document_format.rs](https://github.com/meilisearch/meilisearch/blob/main/meilisearch-lib/src/document_formats.rs).

**Comment by gmourier** [comment]
Hi @MixusMinimax 👋

Sorry for the late answer. Thanks for your feedback.

How would you express this delimiter if you could pass it via the API when sending documents?

Thanks!

**Reply by MixusMinimax** [reply]
For now I don't think it's that important, because I mainly care about `meilisearch-lib`, as I am directly using it. But in the future, `meilisearch-http` could simply use a query parameter, or a request header, whatever you prefer.

I will create a merge request in the following days.

**Reply by gmourier** [reply]
Thanks @MixusMinimax !

**Reply by MixusMinimax** [reply]
I created a pull request: https://github.com/meilisearch/meilisearch/pull/2803
Do I need to create a blank issue in the meilisearch repo, or can I link to this conversation?

**Reply by gmourier** [reply]
Hi @MixusMinimax,

 Sorry for the delay; I've seen the issue associated with the PR. Thanks a lot!

**Comment by gmourier** [comment]
Hey @MixusMinimax 👋

We have planned to add a way to select a CSV delimiter customization in `v1.1`

I'm locking that discussion in favor of https://github.com/meilisearch/product/discussions/541

Thanks!

**Comment by gmourier** [comment]
Hello, everyone following this discussion 👋 

We have just released the first RC (release candidate) of Meilisearch containing this new implementation!

You can test it by using
-  the [release assets](https://github.com/meilisearch/meilisearch/releases)
- the [Meilisearch Docker image](https://hub.docker.com/repository/docker/getmeili/meilisearch/general)

```bash
docker run -it --rm -p 7700:7700 -v $(pwd)/meili_data:/meili_data getmeili/meilisearch:v1.1.0-rc.0
```

You are welcome to communicate any feedback about this new implementation in this discussion.

If you encounter any bugs, please report them [here](https://github.com/meilisearch/meilisearch/issues/new?assignees=&labels=&template=bug_report.md&title=).
Thanks in advance for your help and your involvement in Meilisearch ❤️ 

🎉 Official and stable release containing this change will be available on 3rd April 2023

⚠️ RC (release candidates) are not recommended for production
